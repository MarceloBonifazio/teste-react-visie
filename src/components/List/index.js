import React from 'react';
import uuid from 'uuid/v4';
import PropTypes from 'prop-types';

import Paper from '@material-ui/core/Paper';

import Person from '../Person';
import useStyles from '../styles';

const List = ({ items, removeItem }) => {
  const styles = useStyles();

  if (items.length === 0) {
    return <Paper className={styles.root}>Lista vazia</Paper>;
  }

  return items.map(({ name, eye_color: color, id }) => (
    <Person
      key={uuid()}
      name={name}
      color={color}
      id={id}
      removeItem={removeItem}
    />
  ));
};

List.propTypes = {
  item: PropTypes.arrayOf(PropTypes.shape({})),
  removeItem: PropTypes.func.isRequired,
};

List.defaultProps = {
  item: [],
};

export default List;
