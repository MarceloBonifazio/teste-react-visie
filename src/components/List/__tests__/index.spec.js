import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import List from '../index';

import * as response from '../../../__mocks__/response.json';

const { data } = response;

jest.mock('../../Person', () => ({
  __esModule: true,
  default: () => <div>Person</div>,
}));

const removeItem = jest.fn();

describe('List', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  describe('Should component render correctly...', () => {
    it('Without data', () => {
      const { baseElement, getByText } = render(
        <List removeItem={removeItem} items={[]} />
      );
      expect(baseElement).toMatchSnapshot();
      expect(getByText(/Lista vazia/)).toBeInTheDocument();
    });

    it('With mock response', () => {
      const { baseElement, queryAllByText, queryByText } = render(
        <List removeItem={removeItem} items={data.results} />
      );
      expect(baseElement).toMatchSnapshot();
      expect(queryByText(/Lista vazia/)).not.toBeInTheDocument();
      expect(queryAllByText(/Person/).length).toStrictEqual(10);
    });
  });
});
