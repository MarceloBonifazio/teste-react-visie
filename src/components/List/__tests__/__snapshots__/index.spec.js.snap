// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Should component render correctly... With mock response 1`] = `
<body>
  <div>
    <div>
      Person
    </div>
    <div>
      Person
    </div>
    <div>
      Person
    </div>
    <div>
      Person
    </div>
    <div>
      Person
    </div>
    <div>
      Person
    </div>
    <div>
      Person
    </div>
    <div>
      Person
    </div>
    <div>
      Person
    </div>
    <div>
      Person
    </div>
  </div>
</body>
`;

exports[`List Should component render correctly... Without data 1`] = `
<body>
  <div>
    <div
      class="MuiPaper-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
    >
      Lista vazia
    </div>
  </div>
</body>
`;
